<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ProgramPageGeneral</class>
 <widget class="QWidget" name="ProgramPageGeneral">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>517</width>
    <height>639</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <layout class="QGridLayout" name="gridLayout">
     <item row="5" column="0">
      <widget class="QLabel" name="label_language">
       <property name="text">
        <string>Language:</string>
       </property>
       <property name="buddy">
        <cstring>combo_language</cstring>
       </property>
      </widget>
     </item>
     <item row="5" column="1">
      <widget class="QComboBox" name="combo_language">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
         <horstretch>1</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <item>
        <property name="text">
         <string>Select...</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>C/C++</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>C#</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Delphi/Lazerus</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Java</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>JavaScript (Node.js/Browser)</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Octave</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Perl</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>PHP</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Python</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Ruby</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Shell</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Visual Basic .NET</string>
        </property>
       </item>
      </widget>
     </item>
     <item row="3" column="0">
      <widget class="QLabel" name="label_identifier">
       <property name="text">
        <string>Identifier:</string>
       </property>
       <property name="buddy">
        <cstring>edit_identifier</cstring>
       </property>
      </widget>
     </item>
     <item row="3" column="1">
      <widget class="QLineEdit" name="edit_identifier">
       <property name="maxLength">
        <number>256</number>
       </property>
      </widget>
     </item>
     <item row="4" column="1">
      <widget class="QLabel" name="label_identifier_help">
       <property name="text">
        <string>The identifier is used to uniquely identify a program. Therefore, the same identifier cannot be used for more than one program at the same time. If the specified identifier is already in use, it will be shown in red. Because the identifier will also be used as directory name it can only contain letters [a-zA-Z], numbers [0-9], underscores [_], dots [.] and hyphens [-]. But it cannot begin with a dot or a hyphen and has to be at least three characters long.</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
       </property>
       <property name="wordWrap">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="6" column="1">
      <widget class="QLabel" name="label_language_help">
       <property name="text">
        <string>The programming language affects how the program will be executed.</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
       </property>
       <property name="wordWrap">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="0" column="0">
      <widget class="QLabel" name="label_name">
       <property name="text">
        <string>Name:</string>
       </property>
       <property name="buddy">
        <cstring>edit_name</cstring>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QLineEdit" name="edit_name">
       <property name="maxLength">
        <number>256</number>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="QCheckBox" name="check_auto_generate">
       <property name="text">
        <string>Auto-generate unique identifier</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QLabel" name="label_6">
       <property name="text">
        <string>This is the display name that will be used to refer to the program.</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
       </property>
       <property name="wordWrap">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="7" column="1">
      <widget class="QPlainTextEdit" name="text_description"/>
     </item>
     <item row="7" column="0">
      <widget class="QLabel" name="label_description">
       <property name="text">
        <string>Description:</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
       </property>
       <property name="buddy">
        <cstring>text_description</cstring>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>0</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>edit_name</tabstop>
  <tabstop>check_auto_generate</tabstop>
  <tabstop>edit_identifier</tabstop>
  <tabstop>combo_language</tabstop>
  <tabstop>text_description</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
